src/http/multipart/generator.cr

```diff
@@ -0,0 +1,193 @@
+require "secure_random"
+
+module HTTP::Multipart
+  class GenerationException < Exception
+  end
+
+  # Generates a multipart MIME message.
+  #
+  # ### Example
+  #
+  # ```
+  # io = IO::Memory.new
+  # multipart = HTTP::Multipart::Generator.new(io)
+  # multipart.body_part HTTP::Headers{"Content-Type" => "text/plain"}, "hello!"
+  # multipart.finish
+  # io.to_s # => "----------------------------DTf61dRTHYzprx7rwVQhTWr7\r\nContent-Type: text/plain\r\n\r\nhello!\r\n----------------------------DTf61dRTHYzprx7rwVQhTWr7--"
+  # ```
+  class Generator
+    # Creates a new `Multipart::Generator` which writes the generated multipart
+    # message to *io*, using the multipart boundary *boundary*.
+    def initialize(@io : IO, @boundary = "--------------------------#{SecureRandom.urlsafe_base64(18)}")
+      @state = :START
+    end
+
+    getter boundary
+
+    # Finite State Machine diagram: https://gist.github.com/RX14/221c1edfa98d1196711515d4b5c264eb
+
+    # Generates a content type header with multipart subtype *subtype*, and
```

Comment:
Maybe "Returns"? I initially thought this was appended to the generated content (because it says "generates" and this is a Generator class)